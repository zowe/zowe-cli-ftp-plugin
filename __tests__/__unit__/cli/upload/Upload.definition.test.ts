/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import * as UploadDefinition from "../../../../src/cli/upload/Upload.definition";

describe("Upload definition", () => {

    it("should match the snapshot", () => {
        UploadDefinition.children.forEach((child) => {
            child.handler = child.handler.replace(/^.*[\\/]/, "");
        });
        expect(UploadDefinition).toMatchSnapshot();
    });

});
