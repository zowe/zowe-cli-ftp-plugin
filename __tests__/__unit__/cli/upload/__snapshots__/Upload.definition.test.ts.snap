// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload definition should match the snapshot 1`] = `
Object {
  "aliases": Array [
    "ul",
  ],
  "children": Array [
    Object {
      "aliases": Array [
        "ftds",
      ],
      "description": "Upload contents of a local file to a z/OS data set",
      "examples": Array [
        Object {
          "description": "Upload to \\"ibmuser.cntl(iefbr14)\\" from the file iefbr14.txt",
          "options": "iefbr14.txt \\"ibmuser.cntl(iefbr14)\\"",
        },
        Object {
          "description": "Upload to \\"ibmuser.cntl(iefbr14)\\" from the file iefbr14.txt with the DCB parameters",
          "options": "iefbr14.txt \\"ibmuser.cntl(iefbr14)\\" --dcb \\"RECFM=FB LRECL=326 BLKSIZE=23472\\"",
        },
      ],
      "handler": "FileToDataSet.Handler",
      "name": "file-to-data-set",
      "options": Array [
        Object {
          "aliases": Array [
            "b",
          ],
          "description": "Upload content in binary mode.",
          "name": "binary",
          "type": "boolean",
        },
        Object {
          "aliases": Array [],
          "description": "DCB parameters for sequential dataset (PS dataset) allocation if not existing. It is space separated like RECFM=FB LRECL=326 BLKSIZE=23472",
          "name": "dcb",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "ec",
          ],
          "defaultValue": null,
          "description": "The encoding for download and upload of z/OS data set.",
          "name": "encoding",
          "type": "string",
        },
      ],
      "positionals": Array [
        Object {
          "description": "Upload the contents of this file to the data set",
          "name": "file",
          "required": true,
          "type": "existingLocalFile",
        },
        Object {
          "description": "The data set (PDS member or physical sequential data set) to which you would like to upload content.",
          "name": "dataSet",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "Upload from local file to data set",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "stds",
      ],
      "description": "Upload contents piped to stdin to a z/OS data set",
      "examples": Array [
        Object {
          "description": "Upload to \\"ibmuser.cntl(iefbr14)\\" from standard input (you can pipe into this command)",
          "options": "\\"ibmuser.cntl(iefbr14)\\"",
        },
      ],
      "handler": "StdinToDataSet.Handler",
      "name": "stdin-to-data-set",
      "options": Array [
        Object {
          "aliases": Array [
            "b",
          ],
          "description": "Upload content in binary mode.",
          "name": "binary",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "ec",
          ],
          "defaultValue": null,
          "description": "The encoding for download and upload of z/OS data set.",
          "name": "encoding",
          "type": "string",
        },
      ],
      "positionals": Array [
        Object {
          "description": "The data set (PDS member or physical sequential data set) to which you would like to upload content.",
          "name": "dataSet",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "Upload from stdin to a data set",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "stuf",
        "stu",
      ],
      "description": "Upload from stdin to a Unix System Services File",
      "examples": Array [
        Object {
          "description": "Upload to \\"/u/users/ibmuser/iefbr14.txt\\" from standard input (you can pipe into this command)",
          "options": "\\"/u/users/ibmuser/iefbr14.txt\\"",
        },
      ],
      "handler": "StdinToUssFile.Handler",
      "name": "stdin-to-uss-file",
      "options": Array [
        Object {
          "aliases": Array [
            "b",
          ],
          "description": "Upload content in binary mode.",
          "name": "binary",
          "type": "boolean",
        },
      ],
      "positionals": Array [
        Object {
          "description": "The USS file to which you would like to upload content.",
          "name": "ussFile",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "Upload stdin to a USS File",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "ftuf",
        "ftu",
      ],
      "description": "Upload contents of a local to a Unix System Services file.",
      "examples": Array [
        Object {
          "description": "Upload to \\"/u/users/ibmuser/iefbr14.txt\\" from the file iefbr14.txt",
          "options": "iefbr14.txt \\"/u/users/ibmuser/iefbr14.txt\\"",
        },
      ],
      "handler": "FileToUssFile.Handler",
      "name": "file-to-uss-file",
      "options": Array [
        Object {
          "aliases": Array [
            "b",
          ],
          "description": "Upload content in binary mode.",
          "name": "binary",
          "type": "boolean",
        },
      ],
      "positionals": Array [
        Object {
          "description": "Upload the contents of this local file to a data set.",
          "name": "file",
          "required": true,
          "type": "existingLocalFile",
        },
        Object {
          "description": "The USS file to which you would like to upload content.",
          "name": "ussFile",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "Upload a local file to a USS File",
      "type": "command",
    },
  ],
  "description": "Upload data set and USS content",
  "name": "upload",
  "passOn": Array [
    Object {
      "ignoreNodes": Array [
        Object {
          "type": "group",
        },
      ],
      "merge": true,
      "property": "options",
      "value": Array [
        Object {
          "aliases": Array [
            "H",
          ],
          "description": "The hostname or IP address of the z/OS server to connect to.",
          "group": "FTP Connection options",
          "name": "host",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "P",
          ],
          "defaultValue": 21,
          "description": "The port of the z/OS FTP server.",
          "group": "FTP Connection options",
          "name": "port",
          "type": "number",
        },
        Object {
          "aliases": Array [
            "u",
          ],
          "description": "Username for authentication on z/OS",
          "group": "FTP Connection options",
          "name": "user",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "p",
            "pass",
            "pw",
          ],
          "description": "Password to authenticate to FTP.",
          "group": "FTP Connection options",
          "name": "password",
          "type": "string",
        },
        Object {
          "defaultValue": true,
          "description": "Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990). Note: Unfortunately, this plugin's functionality only works with FTP and FTPS, not 'SFTP' which is FTP over SSH.",
          "group": "FTP Connection options",
          "name": "secure-ftp",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "ru",
          ],
          "defaultValue": null,
          "description": "Reject self-signed certificates. Only specify this if you are connecting to a secure FTP instance.",
          "group": "TLS / Secure Connection options",
          "name": "reject-unauthorized",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "sn",
          ],
          "defaultValue": null,
          "description": "Server name for the SNI (Server Name Indication) TLS extension. Only specify if you are connecting securely",
          "group": "TLS / Secure Connection options",
          "name": "server-name",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "ct",
          ],
          "defaultValue": 10000,
          "description": "How long (in milliseconds) to wait for the control connection to be established.",
          "group": "FTP Connection options",
          "name": "connection-timeout",
          "type": "number",
        },
      ],
    },
  ],
  "summary": "Upload data set and USS content",
  "type": "group",
}
`;
